### PR TITLE
Hotfix

### PR DESCRIPTION
**Pull request by JediSebas**

### CODE REVIEW

File: src/main/java/com/jedisebas/restapi/constants/TestDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.jedisebas.restapi.entity.PersonalDetails;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TestDataProvider {
+
+    public static PersonalDetails createProperlyPersonalDetails() {
+        return PersonalDetails.builder()
+                .firstName("Szymon")
```
Comment (yarpo) [inline src/main/java/com/jedisebas/restapi/constants/TestDataProvider.java:13]: Extract this "magic strings" into consts.

File: src/main/java/com/jedisebas/restapi/service/AnonymizeService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        return lastName.charAt(0) + "*".repeat(lastName.length() - 1);
+    }
+
+    public static String anonymizeEmail(final String email) {
+        final StringBuilder emailBuilder = new StringBuilder();
+        emailBuilder.append(email.charAt(0));
+        emailBuilder.append("***");
+
+        int i = 4;
+        while (email.charAt(i + 1) != '@') {
```
Comment (yarpo) [inline src/main/java/com/jedisebas/restapi/service/AnonymizeService.java:20]: why not split string by `@`. You get an array with 2 elements. The second is domain.

File: src/main/java/com/jedisebas/restapi/service/AnonymizeService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            i++;
+        }
+
+        for (int j = 0; j < 3; j++) {
+            emailBuilder.append(email.charAt(i));
```
Comment (yarpo) [inline src/main/java/com/jedisebas/restapi/service/AnonymizeService.java:25]: Why not?
text. substring(start, length);

?

File: src/main/java/com/jedisebas/restapi/service/ValidatorService.java [hunk 4]
```diff
@@ -9,6 +9,11 @@
 public class ValidatorService {
 
     public void extracted(final PersonalDetailsDto personalDto) {
```
Comment (yarpo) [inline src/main/java/com/jedisebas/restapi/service/ValidatorService.java:11]: name is not very self-describing.

File: src/main/java/com/jedisebas/restapi/service/AnonymizeService.java [hunk 5]
```diff
@@ -12,35 +12,27 @@ public static String anonymizeLastName(final String lastName) {
     }
 
     public static String anonymizeEmail(final String email) {
```
Comment (yarpo) [inline src/main/java/com/jedisebas/restapi/service/AnonymizeService.java:14]: add unit tests, please

Comment (JediSebas) [reply src/main/java/com/jedisebas/restapi/service/AnonymizeService.java:14]: Ok, I'll do

File: src/test/java/com/jedisebas/restapi/service/AnonymizeServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class AnonymizeServiceTest {
+
+    @Test
+    void givenLastName_whenAnonymized_thenReturnResult() {
+        assertEquals("A***", AnonymizeService.anonymizeLastName("Aqwe"));
```
Comment (yarpo) [inline src/test/java/com/jedisebas/restapi/service/AnonymizeServiceTest.java:11]: what about:
- empty string
- null
- string that starts with (one or more)  whitespace(s)
- string that ends with (one or more) whitespace(s)

File: src/test/java/com/jedisebas/restapi/service/AnonymizeServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        assertEquals("P*****", AnonymizeService.anonymizeLastName("Pqwert"));
+        assertEquals("V", AnonymizeService.anonymizeLastName("V"));
+        assertEquals("q*****", AnonymizeService.anonymizeLastName("qwerty"));
+        assertEquals("**", AnonymizeService.anonymizeLastName("**"));
+    }
+
+    @Test
+    void givenEmail_whenAnonymized_thenReturnResult() {
+        assertEquals("a***a@a***a.a", AnonymizeService.anonymizeEmail("a@a.a"));
```
Comment (yarpo) [inline src/test/java/com/jedisebas/restapi/service/AnonymizeServiceTest.java:20]: what about:
sebastian.dolata@zst.edu.pl?

Comment (yarpo) [inline src/test/java/com/jedisebas/restapi/service/AnonymizeServiceTest.java:20]: this is a good edge case - is this correct?

File: src/test/java/com/jedisebas/restapi/service/AnonymizeServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        assertEquals("*****@*****.*", AnonymizeService.anonymizeEmail("*@*.*"));
```
Comment (yarpo) [inline src/test/java/com/jedisebas/restapi/service/AnonymizeServiceTest.java:21]: I think it's invalid email.

**Review by yarpo — APPROVED** [review]
